Content.Server/Zombies/ZombieSystem.cs

```diff
@@ -62,18 +65,32 @@ public override void Update(float frameTime)
             base.Update(frameTime);
             var query = EntityQueryEnumerator<PendingZombieComponent>();
             var curTime = _timing.CurTime;
+            var stateQuery = GetEntityQuery<MobStateComponent>();
 
             // Hurt the living infected
             while (query.MoveNext(out var uid, out var comp))
             {
                 if (comp.NextTick + TimeSpan.FromSeconds(1) > curTime)
                     continue;
 
+                comp.NextTick = curTime;
+
                 comp.InfectedSecs += 1;
+                // See if there should be a warning popup for the player.
+                if (comp.InfectionWarnings.TryGetValue(comp.InfectedSecs, out var popupStr))
+                {
+                    _popup.PopupEntity(Loc.GetString(popupStr), uid, uid);
+                }
+
+                if (comp.InfectedSecs < 0)
+                {
+                    // This zombie has a latent virus, probably set up by ZombieRuleSystem. No damage yet.
+                    continue;
+                }
```

Comment:
does this check ever get passed? we always update InfectedSecs just before this and since it starts at 0 and nothing subtracts from it is should always be above 0

Reply:
Yes, latent virus, as per the comment.